src/main/java/org/kohsuke/github/Requester.java

```diff
@@ -497,73 +497,81 @@ public InputStream fetchStream() throws IOException {
     }
 
     private <T> T _fetch(String tailApiUrl, URL url, SupplierThrows<T, IOException> supplier) throws IOException {
-        while (true) {// loop while API rate limit is hit
+        int responseCode = -1;
+        String responseMessage = null;
+
+        int retries = CONNECTION_ERROR_RETRIES;
+
+        do {
+            // if we fail to create a connection we do not retry and we do not wrap
+            uc = null;
```

Comment:
This ensures that on retry we don't somehow accidentally use the connection from the previous loop.